test/integration_test.go

```diff
@@ -551,11 +554,119 @@ func (ts *IntegrationTestSuite) TestNonDeterministicWorkflowQuery() {
 
 func (ts *IntegrationTestSuite) TestOverrideSpanContext() {
 	var result map[string]string
-	err := ts.executeWorkflow("test-override-span-context", ts.workflows.OverrideSpanContext, &result)
+	_, err := ts.executeWorkflow("test-override-span-context", ts.workflows.OverrideSpanContext, &result)
 	ts.NoError(err)
 	ts.Equal("some-value", result["mockpfx-baggage-some-key"])
 }
 
+// beforeVersionedWorkflowTest registers appropriate versioned workflow and activity to emulate the versioned workflow test.
+func (ts *IntegrationTestSuite) beforeVersionedWorkflowTest(testName string, w worker.Worker) {
+	switch testName {
+	case "TestVersionedWorkflowV1":
+		replaytests.SetupWorkerForVersionedWorkflowV1(w)
+	case "TestVersionedWorkflowV2":
+		replaytests.SetupWorkerForVersionedWorkflowV2(w)
+	case "TestVersionedWorkflowV3":
+		replaytests.SetupWorkerForVersionedWorkflowV3(w)
+	case "TestVersionedWorkflowV4":
+		replaytests.SetupWorkerForVersionedWorkflowV4(w)
+	case "TestVersionedWorkflowV5":
+		replaytests.SetupWorkerForVersionedWorkflowV5(w)
+	case "TestVersionedWorkflowV6":
+		replaytests.SetupWorkerForVersionedWorkflowV6(w)
+	}
+}
```

Comment:
just put these inline, only a single test cares about each individual value.  no need to add a shared switch statement that every test runs.

Reply:
Didn't know that the worker can register workflow after its start, so removed it